### PR TITLE
[ISSUE #3506] removed return statement from the for loop

### LINKED ISSUE #3506: [Enhancement] The return is used directly in the loop.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

image
located at:
eventmesh-registry-plugin/eventmesh-registry-etcd/src/main/java/org/apache/eventmesh/registry/etcd/service/EtcdCustomService.java line 96

analysis and explanation:
A loop with at most one iteration is equivalent to the use of an if statement to conditionally execute one piece of code. No developer expects to find such a use of a loop statement.

### Describe the solution you'd like

e.g.
image

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by PickBas**
Fixes #3506.

### Modifications

Removed return statement from the for loop

### Documentation

- Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Review by Alonexc — DISMISSED** [review]
LGTM

**Review by mytang0 — CHANGES_REQUESTED** [review]
Modifications result in logic changes: the original logic is to end after parsing the first one.

image

**Comment by xwm1992** [comment]
52148992-72b1-421a-9463-1d23d9c975ed
I haven confirmed with the author, for now there is only one result under the loop, it's ok.

**Review by xwm1992 — APPROVED** [review]

**Comment by PickBas** [comment]
@mytang0 Are changes still requested or it's ok?

**Comment by mytang0** [comment]
> @mytang0 Are changes still requested or it's ok?

Hello @PickBas , can you help to optimize it? Remove the for loop and add a comment. Grateful.

**Comment by PickBas** [comment]
@mytang0 have a look please

File: eventmesh-registry-plugin/eventmesh-registry-etcd/src/main/java/org/apache/eventmesh/registry/etcd/service/EtcdCustomService.java [hunk 1]
```diff
@@ -77,31 +77,29 @@ public List<EventMeshServicePubTopicInfo> findEventMeshServicePubTopicInfos() th
...
-                        JsonUtils.parseObject(new String(kv.getValue().getBytes(), Constants.DEFAULT_CHARSET), EventMeshAppSubTopicInfo.class);
+                EventMeshAppSubTopicInfo eventMeshAppSubTopicInfo =
+                    JsonUtils.parseObject(
+                        new String(keyValues.get(0).getValue().getBytes(), Constants.DEFAULT_CHARSET),
+                        EventMeshAppSubTopicInfo.class
+                    );
+                if (eventMeshAppSubTopicInfo != null) {
                     return eventMeshAppSubTopicInfo;
+                } else {
+                    throw new NullPointerException("eventMeshAppSubTopicInfo is null!");
```
Comment (mytang0) [inline eventmesh-registry-plugin/eventmesh-registry-etcd/src/main/java/org/apache/eventmesh/registry/etcd/service/EtcdCustomService.java:96]: Return directly, there is already a null check outside.

Comment (PickBas) [reply eventmesh-registry-plugin/eventmesh-registry-etcd/src/main/java/org/apache/eventmesh/registry/etcd/service/EtcdCustomService.java]: Done

**Review by mytang0 — APPROVED** [review]
LGTM

**Pull request merged**